subsys/net/l2/openthread/openthread.c

```diff
@@ -360,8 +415,11 @@ static void openthread_start(struct openthread_context *ot_context)
 	} else if (IS_ENABLED(CONFIG_OPENTHREAD_JOINER_AUTOSTART)) {
 		/* No dataset - initiate network join procedure. */
 		NET_DBG("Starting OpenThread join procedure.");
-
-		error = otJoinerStart(ot_instance, OT_JOINER_PSKD, NULL,
+		// should this fail, we'll stick to OT_JOINER_PSKD
+		(void) devicedata_get_pskd(&pPSKD);
```

Comment:
If the result is ignored anyway, I think the function should not return the result in a first place (i. e. be of `void` type).

Reply:
maybe the return value of the app-defined function could be used elsewhere, but it is not interesting here?